Create a forgotten favorites feature

1hitsong opened the pull request:
Creates a list of tracks weighted by play count and time since last play.

Screenshot of feature
![image](https://github.com/user-attachments/assets/ec59b65c-85ea-41b2-87cb-d5ccd4d069ac)

Fixes #18

Chaphasilor commented:
Nice, that looks promising. I think it would be neat to emphasize the "favorites" aspect of this a bit more, only including tracks that we're played a lot, or are actual favorites.  
From a quick glance it seems like currently any track that was played at least once can be included, and it's ordered by when it was last played. So if I played a track from a new album at the beginning of the year, decided I didn't like it, and never played it again, it would show up right at the top of the "forgotten favorites", which seems a bit counter-intuitive. 

What do you think?

1hitsong commented:
> So if I played a track from a new album at the beginning of the year, decided I didn't like it, and never played it again, it would show up right at the top of the "forgotten favorites", which seems a bit counter-intuitive.

Only if you didn't play much else during the year. I have a lot of tracks I only played once at the start of the year, and you can see it only picked up the ones I played at least twice.

The sort weights the play count. We can play with the weight value to fine tune the formula.

Chaphasilor commented:
Ahh yeah I didn't see that.  

Another problem I could see is that if there really isn't a "forgotten love", it would still create and show the list. That could contain something like "last played December 20th", in theory. Maybe we should add a threshold there?  
The skip features have a similar issue.

1hitsong commented:
I've added a minimum last play age - currently set to 180 days, roughly half a year.

I also added code to auto the forgotten favorites slide if no tracks are found. I coded it so we can easily use the auto skip logic for other feature slides.

Chaphasilor reviewed (commented):
Thanks for the minimum age and auto skip. I think we can set the age lower, 3-4 months are probably enough.  
Also, I'd like to stay consistent with the other features and only feature the first 5 tracks, with the remaining tracks in a plain list, to avoid scrolling an improve screenshotability on mobile. Could you please change that?  
Also, you should add a padding to the description/subtitle on the feature (or limit the width in another way), so it can't stretch to the edges of the screen, as that looks a bit weird. Here's a picture:  
![image](https://github.com/user-attachments/assets/c24c9b07-5405-4168-b4f2-14735e62e5c2)

Chaphasilor commented:
Sorry about the force push. Commit ended up in the wrong branch :(

Chaphasilor commented:
Since all other features (top albums, top tracks, etc.) also list the top 20, and feature the first 5, I've adjusted that quickly.  
Theoretically this can be merged, but do you think it would be possible to somehow integrate this with the `getTopItems()` in `aggregate.js`? This is how all other top-something features are generated, including most/least skipped. So it would be nice if the "forgotten favorite" could stay consistent with this.

If it's more trouble than benefit, we can also leave it like this :)

1hitsong commented:
> Maybe we could have some sort of global configuration/constant object that each function can then use to determine these things? What do you think?

🤘 Heck yeah, I'm in for that! I'll open a ticket.

1hitsong commented:
Once we get this feature buttoned up and merged, I'll go ahead and work on ticket #42 next before working on any new stuff. As part of #42, I'll update not only this feature, but also the others to use the new global config.

1hitsong commented:
@Chaphasilor Anything else you need on this PR to get it merged?
